syntax = "proto3";

package protocol.payment.deduction.v1;

// The greeting service definition.
service DeductionAdminApi {
  rpc GetDeductionList(DeductionAdminReq) returns (DeductionAdminResp) {}
  rpc AdminOp(DeductionAdminReq) returns (DeductionAdminResp) {}
}
message DeductionStruct {
  string title = 1;
  string desc = 2;
  uint32 status = 3;
  uint32 id = 4;
  uint32 pay_method_id = 5;
  uint32 detail_id = 6;
}

message DeductionAdminResp {
  bool success = 1;
  repeated DeductionStruct data = 2;
  string message = 3;
}

message DeductionAdminReq {
  DeductionStruct data = 1;
}
